CMakeLists.txt

```
@@ -48,6 +48,32 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 endif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 # End HeadSpin Configuration
 
+# Custom configuration for Release (optimized) with PDBs
+if(MSVC)
+    if(NOT "RelWithPDB" IN_LIST CMAKE_CONFIGURATION_TYPES)
+        set(CMAKE_CONFIGURATION_TYPES "${CMAKE_CONFIGURATION_TYPES};RelWithPDB" CACHE STRING "" FORCE)
+    endif()
+
+    # Yeah, this looks strange. The semicolon is to establish an empty list item
+    # for the configuration-less import.
+    set(CMAKE_MAP_IMPORTED_CONFIG_RELWITHPDB "Release;")
+
+    # Need /Zi for the PDB
+    set(CMAKE_C_FLAGS_RELWITHPDB   "/MD /Zi /O2 /Ob2 /DNDEBUG" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS_RELWITHPDB "/MD /Zi /O2 /Ob2 /DNDEBUG" CACHE STRING "")
```

Maybe it would be more future-proof to say
```suggestion
    set(CMAKE_C_FLAGS_RELWITHPDB   "${CMAKE_C_FLAGS_RELEASE} /Zi" CACHE STRING "")
    set(CMAKE_CXX_FLAGS_RELWITHPDB "${CMAKE_CXX_FLAGS_RELEASE} /Zi" CACHE STRING "")
```